### DISCUSSION TITLE
Run jobs from different projects in series

### DISCUSSION THREAD

**Discussion by tiagodj** [Escape-Technologies/mookme#103] [Q&A] [answered]
I have a mono repo with 2 projects (backend and front end) which have mookme hooks setup.

When we make changes to both projects at the same time, mookme triggers both hooks in parallel, which cause one of them to fail.

This happens because the tool `lint-staged` (triggered in the hook step) uses a local temporary storage and that causes conflicts when 2 processes are running in parallel.

```
 Hooks : services/apiserver-graphql  Done ✓ 
→ Lint & Format staged files > yarn lint-staged 
✅ Done. 

 Hooks : services/client  Error × 
→ Lint & Format staged files > yarn lint-staged 
❌ Error. 

Hook of package services/client failed at step Lint & Format staged files 

[FAILED] lint-staged automatic backup is missing!
```

I know about the `serial` option, but that is to serialize steps on the same project. How about serializing steps among different projects? Would it be possible?

Thanks

**Comment by LMaxence** [comment, marked as answer]
Hello !

From what I understand, `lint-staged`, due to this behaviour does not properly run simultaneously in the same monorepo. I can provide the following options for you in this situation:

1) There is actually an option for lint stage to checkout and stash only a specific workspace while running ? There might already be related issues like that out there

2) You can put both of the lint jobs in a hooks file above (in `services` in your situation). Thus, both jobs are in the same hook file and can become `serial`.

3) Instead of linting both workspace separately, you might only lint them once by configuering the linting task at your monorepo level. [I wrote a bit about such a setup on Escape's engineering blog](https://escape.tech/blog/code-quality-nodejs-monorepo/)

Solution 2) is just a workaround I would not hold it as an ideal solution, but I can't be sure that 1) or 3) are achievable in your project

Hope that helps !
